File: infisocial_flutter/lib/models/post_model.dart
enum MediaType { image, video }

class PostModel {
  final String postId;
  final String content;
  final String mediaType;
  final String mediaUrl;
  final List<String> likes;
  final List<String> comments;
  final String postedBy;
  final String postOwnerUsername;
  final String? postOwnerAvatar;

  const PostModel({
    this.postId = '',
    this.content = '',
    this.mediaType = 'image',
    required this.likes,
    required this.comments,
    required this.mediaUrl,
    required this.postedBy,
    required this.postOwnerUsername,
    this.postOwnerAvatar,
  });

  // Factory method to create a Post from a Map (useful for Firebase or API responses)
  factory PostModel.fromJson(Map<String, dynamic> json) {
    return PostModel(
      postId: json['_id'] ?? '',
      content: json['content'] ?? '',
      mediaType: json['mediaType'] ?? 'image',
      mediaUrl: json['mediaUrl'] ?? '',
      likes: List<String>.from(json['likes']),
      comments: List<String>.from(json['comments']),
      postedBy: json['postedBy']['_id'] ?? '',
      postOwnerUsername: json['postedBy']['username'] ?? 'username',
      postOwnerAvatar: json['postedBy']['avatarUrl'],
    );
  }

  // Convert a Post object to a Map (useful for Firestore or API requests)
  Map<String, dynamic> toJson() {
    return {
      '_id': postId,
      'content': content,
      'mediaType': mediaType,
      'mediaUrl': mediaUrl,
      'likes': likes,
      'comments': comments,
      'postedBy': postedBy,
    };
  }
}

File: infisocial_node/layers/utils/models/post.model.js
const mongoose = require("mongoose");

const postSchema = new mongoose.Schema({
    postedBy: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        required: true,
    },
    content: {
        type: String,
    },
    mediaUrl: {
        type: String,
    },
    mediaType: {
        type: String,
        enum: ["image", "video"],
    },
    tags: {
        type: [String],
        default: [],
        index: true,
    },
    taggedUsers: {
        type: [mongoose.Schema.Types.ObjectId],
        ref: 'User',
        default: [],
        index: true,
      },
    likes: {
        type: [mongoose.Schema.Types.ObjectId],
        ref: "User",
        default: [],
    },
    comments: {
        type: [mongoose.Schema.Types.ObjectId],
        ref: "Comment",
        default: [],
    },
    isActive: {
        type: Boolean,
        default: true
    }
},
    { timestamps: true }
);

const Post = mongoose.model("Post", postSchema);

module.exports = Post;
